src/hooks/__internal__/useCharacterCount/index.spec.js

```diff
@@ -0,0 +1,111 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import { mount } from "enzyme";
+
+import { assertStyleMatch } from "../../../__spec_helper__/test-utils";
+import CharacterCount from "../../../__internal__/character-count";
+import I18nProvider from "../../../components/i18n-provider";
+import useCharacterCount from ".";
+
+const TestComponent = ({
+  value,
+  characterLimit,
+  warnOverLimit,
+  enforceCharacterLimit,
+}) => {
+  const [maxLength, characterCount] = useCharacterCount(
+    value,
+    characterLimit,
+    warnOverLimit,
+    enforceCharacterLimit
+  );
+
+  return (
+    <>
+      {characterCount}
+      <span data-element="max-length">{maxLength}</span>
+    </>
+  );
+};
+
+const render = (props, params) => mount(<TestComponent {...props} />, params);
+
+TestComponent.propTypes = {
+  value: PropTypes.string,
+  characterLimit: PropTypes.string,
+  warnOverLimit: PropTypes.bool,
+  enforceCharacterLimit: PropTypes.bool,
+};
+
+describe("useCharacterCount", () => {
+  it("returns a counter and no maxLength", () => {
+    const wrapper = render({
+      value: "test string",
```

Comment:
I think that we can assign it to the default prop in Mock Component as we are using it always the same

Reply:
I think now it's more readable. It's just a unit test, not a component.